src/main/java/org/spongepowered/common/effect/sound/SpongeSoundBuilder.java

```diff
@@ -24,26 +24,16 @@
  */
 package org.spongepowered.common.effect.sound;
 
-import static com.google.common.base.Preconditions.checkNotNull;
-
 import net.minecraft.util.ResourceLocation;
 import net.minecraft.util.SoundEvent;
+import org.spongepowered.api.CatalogKey;
 import org.spongepowered.api.effect.sound.SoundType;
+import org.spongepowered.common.util.SpongeCatalogBuilder;
 
-public class SpongeSoundBuilder implements SoundType.Builder {
-
-    @Override
-    public SoundType.Builder from(SoundType value) {
-        return this;
-    }
+public final class SpongeSoundBuilder extends SpongeCatalogBuilder<SoundType, SoundType.Builder> implements SoundType.Builder {
 
     @Override
-    public SoundType.Builder reset() {
-        return this;
+    protected SoundType build(CatalogKey key) {
```

Comment:
checkNotNull

Reply:
Doesn't SpongeCatalogBuilder (which I do extend) already null check the key?

Reply:
No cause you don't call super.